tests/unit/logic/vote.spec.js

```diff
@@ -0,0 +1,1012 @@
+			var clock, trs, callback, delegates, rounds, system
+
+			beforeEach(function () {
+				callback = sinon.spy()
+				clock = sinon.useFakeTimers()
+				Vote.__set__('setImmediate', setImmediate)
+				trs = {senderPublicKey: 'foo', asset: {votes: []}}
+				delegates = {}
+				rounds = {}
+				system = {}
+				vote.bind(delegates, rounds, system)
+				Vote.__set__({ modules: {
+					delegates: {
+						checkUnconfirmedDelegates: function (trs, votes, cb) {
+							return setImmediate(cb)
+						}
+					}
+				}
+				})
+				vote.checkUnconfirmedDelegates(trs, callback)
+			})
+
+			afterEach(function () {
+				clock.restore()
+			})
+
+			it('should call to callback without errors', function () {
+				clock.runAll()
+				expect(callback.called).to.be.true
+				expect(callback.args[0][0]).to.equals(undefined)
+			})
+		})
+
+		describe('process()', function () {
+
+			var clock, trs, callback
+
+			beforeEach(function () {
+				trs = 1
+				callback = sinon.spy()
+				clock = sinon.useFakeTimers()
+				Vote.__set__('setImmediate', setImmediate)
+				vote.process(trs, null, callback)
+			})
+
+			it('should call to callback passing the same transaction parameter received', function () {
+				clock.runAll()
+				expect(callback.called).to.be.true
+				expect(callback.args[0][1]).to.equals(trs)
+			})
+
+		})
+
+		describe('getBytes()', function () {
+
+			var trs, result, getBytes
+
+			context('if the input data is wrong', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: 123}}
+					getBytes = function () {
+						vote.getBytes(trs)
+					}
+				})
+
+				it('should throws an error', function () {
+					expect(getBytes).to.throw()
+				})
+			})
+
+			context('if trs.asset.votes is undefined', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: undefined}}
+					result = vote.getBytes(trs)
+				})
+
+				it('should return null', function () {
+					expect(result).to.equals(null)
+				})
+			})
+
+			context('if everything is ok', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: ['+37b6d3fc89fb418fa5e3799e760cc2b1733a567daebb08a531d3d7e8b24f2d22','+9f59d4260dcd848f71d17824f53df31f3dfb87542042590554419ff40542c55e']}}
+					result = vote.getBytes(trs)
+				})
+
+				it('should return a Buffer object', function () {
+					expect(result).to.be.an.instanceof(Buffer)
+				})
+			})
+		})
+
+		describe('apply()', function () {
+			var clock, trs, block, sender, callback, scope, delegates, rounds, system
+
+			beforeEach(function () {
+				callback = sinon.spy()
+				clock = sinon.useFakeTimers()
+				scope = { account: { merge: function (address, data, cb) { cb() } } }
+				rounds = {calc: function (height) { return height }}
+				trs = {asset: {votes: []}}
+				sender = {address: 123}
+				block = {id: 123, height: 123}
+				Vote.__set__('setImmediate', setImmediate)
+				Vote.__set__('self.checkConfirmedDelegates', function (trs, cb) {
```

Comment:
stub this. In this way, you will be able to restore the original function and you'll also be able to spy if it has been properly called or not...

Reply:
:+1:

Reply:
Done!